src/material/menu/menu-content.ts

```diff
@@ -39,7 +41,8 @@ export class MatMenuContent implements OnDestroy {
     private _appRef: ApplicationRef,
     private _injector: Injector,
     private _viewContainerRef: ViewContainerRef,
-    @Inject(DOCUMENT) private _document: any) {}
+    @Inject(DOCUMENT) private _document: any,
+    @Optional() private _changeDetectorRef: ChangeDetectorRef) {}
```

Comment:
The `@Optional` won't help with backwards compatibility. It should be marked as an optional parameter (`_changeDetectorRef?: ChangeDetectorRef`). Also I think we can drop the `@Optional` since we'll always have a `ChangeDetectorRef`.